src/cfn/template.yaml

```diff
@@ -456,6 +356,340 @@ Resources:
       SubnetId: !Ref PublicSubnet3
       NetworkAclId: !Ref PublicNetworkAcl
 
+  DynamoDBEndpoint:
+    Type: AWS::EC2::VPCEndpoint
+    Properties:
+      RouteTableIds:
+        - !Ref PrivateRouteTable1
+        - !Ref PrivateRouteTable2
+        - !If [ HasThreeAZs, !Ref PrivateRouteTable3, !Ref 'AWS::NoValue' ]
+      ServiceName: !Sub 'com.amazonaws.${AWS::Region}.dynamodb'
+      VpcId: !Ref VPC
+
+  S3Endpoint:
+    Type: AWS::EC2::VPCEndpoint
+    Properties:
+      RouteTableIds:
+        - !Ref PrivateRouteTable1
+        - !Ref PrivateRouteTable2
+        - !If [ HasThreeAZs, !Ref PrivateRouteTable3, !Ref 'AWS::NoValue' ]
+      ServiceName: !Sub 'com.amazonaws.${AWS::Region}.s3'
+      VpcId: !Ref VPC
+
+  # Custom Resources
+
+  LambdaSetup:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: MediaAnalysisSetup
+      Runtime: nodejs8.10
+      Handler: index.handler
+      CodeUri: ../backend/functions/setup/
+      Description: Custom Lambda resource for the Media Analysis Cloudformation Stack
+      MemorySize: 128
+      Timeout: 30
+      Role: !GetAtt CustomResourceRole.Arn
+
+  CustomResourceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: MediaAnalysis-setup-S3-fc
+          PolicyDocument:
+            Statement:
+            - Effect: Allow
+              Action: 's3:*'
+              Resource: '*'
+        - PolicyName: MediaAnalysis-cloudwatch-logs
+          PolicyDocument:
+            Statement:
+            - Effect: Allow
+              Action: 'logs:*'
+              Resource: 'arn:aws:logs:*:*:*'
+        - PolicyName: MediaAnalysis-setup-ecr
+          PolicyDocument:
+            Statement:
+            - Effect: Allow
+              Action: 'ecr:DeleteRepository'
+              Resource: '*'
+
+  SetupWebUIAndDockerImage:
+    Type: Custom::Setup
+    Properties:
+      ServiceToken: !GetAtt LambdaSetup.Arn
+      Region: !Ref AWS::Region
+
+  # Computing & Containers
+
+  ECSCluster:
+    Type: AWS::ECS::Cluster
+
+  ECRRepository:
+    Type: AWS::ECR::Repository
+
+  CodeBuildArtifactBucket:
```

Comment:
The code build needs to have a versioned bucket. We may consolidate the webui and codebuild buckets later on, at this stage I decided to separate them.